components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/keyidprovider/KeyIDProvider.java

```diff
@@ -37,4 +38,22 @@ public interface KeyIDProvider {
      */
     public String getKeyId(Certificate certificate, JWSAlgorithm signatureAlgorithm, String tenantDomain)
             throws IdentityOAuth2Exception;
+
+    /**
+     * Method containing the Key ID calculation logic with certificate thumbprint using SHA-1 alg and
+     * signature algorithm.
+     * This method is added to support migration from versions below IS 5.10 as we have switched the hashing algorithm
+     * from SHA-1 to SHA-256.
+     *
+     * @param certificate        Signing Certificate.
+     * @param signatureAlgorithm Signature Algorithm as a String.
+     * @param tenantDomain       Tenant Domain.
+     * @return String value of Previous Key ID.
+     * @throws IdentityOAuth2Exception When failed to generate Key ID properly.
+     */
+    default String getPreviousKeyId(Certificate certificate, JWSAlgorithm signatureAlgorithm, String tenantDomain)
```

Comment:
Just a though, should we really introduce this method to an interface of keyid provider?

If versions below 5.10.0 are not supported at a point, this method will be not useful.

If this is something just for migration, can we check if there is a better way to do this without altering the interface